consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/ExecutionEventsController.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package za.co.absa.spline.consumer.rest.controller
+
+import java.lang.System.currentTimeMillis
+
+import io.swagger.annotations.{Api, ApiOperation, ApiParam}
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation._
+import za.co.absa.spline.consumer.service.model._
+import za.co.absa.spline.consumer.service.repo.ExecutionEventRepository
+
+import scala.concurrent.Future
+
+@RestController
+@Api(tags = Array("execution-events"))
+class ExecutionEventsController @Autowired()(val repo: ExecutionEventRepository) {
+
+  import scala.concurrent.ExecutionContext.Implicits._
+
+  @GetMapping(Array("/execution-events"))
+  @ApiOperation(
+    value = "Get execution events",
+    notes = "Returns a pageable list of execution events filtered by the query parameters",
+    response = classOf[PageableExecutionEventsResponse]
+  )
+  def executionEvents(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The number of parameters should not exceed 8](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4536902)